ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java

```diff
@@ -151,6 +155,10 @@ public void afterPropertiesSet() throws Exception {
 		}
 		server.setTransports(transport);
 		start();
+
+		SocketAcceptor socketAcceptor = server.getSocketAcceptor(transport);
+		InetSocketAddress localAddress = socketAcceptor.getLocalAddress();
+		this.localPort = localAddress.getPort();
```

Comment:
I think this logic should probably be moved to the `start` method since a user could invoke `start` directly.

Reply:
Should we also move the `server` initialization to the `start` method?

Reply:
Let's try and make as little changes in this PR as possible. If we see additional room for improvement, let's create separate tickets.

Reply:
Done.